test/files/run/t6628.scala

```diff
@@ -1,5 +1,5 @@
 object Test {
-  def coll = new Iterable[String] {
+  def coll = new collection.AbstractIterable[String] {
```

Comment:
Why can’t we extend `Iterable` anymore?

Reply:
Iterable (IterableOps) has now tow more abstract methods `fromSpecific` and `iterableFactory`, because `Iterable` doesn't extend `IterableFactoryDefaults`.

`AbstractIterable` on the other hand mixes in `IterableFactoryDefaults`, so it has a single abstract method `iterator`. So it's an easy way to create an Iterable implementation.

Reply:
I see. Should we recommend mixing `AbstractXxx` or `XxxFactoryDefaults` when implementing custom collection types? https://docs.scala-lang.org/overviews/core/custom-collections.html

Reply:
Using `AbstractX` is probably better since it also avoids mixin forwarders, no? I also have to write down a good explanation why we can't mix in `IterableFactoryDefaults` to `Iterable`.

Reply:
We can actually go back to `Iterable` now, Adriaan made it work to have `Iterable` extend `IterableFactoryDefaults`.